hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -969,6 +969,11 @@
   public static final String HBASE_REGION_SPLIT_POLICY_KEY =
     "hbase.regionserver.region.split.policy";
 
+  public static final String RS_PARALLEL_GET_ENABLED = "hbase.server.parallel.get.enabled";
+  public static final String RS_PARALLEL_GET_THREADS = "hbase.server.parallel.get.threads";
```

Comment:
How do I get feedback on how many threads I should run with? What do I look at? Is there a log message or a metric I can see?

Reply:
How does this bloat up our thread count? Already the RegionServer has too many. How many threads will this add?

Reply:
> How do I get feedback on how many threads I should run with? What do I look at? Is there a log message or a metric I can see?

How about add an ExecutorStatusChore to collect the Executor metrics, this is useful for all the Executors opened by RS

Reply:
> How does this bloat up our thread count? Already the RegionServer has too many. How many threads will this add?

Will add 20 threads by default, can be configure with hbase.server.parallel.get.threads

Reply:
Stats on executors for general RS sounds good. Need feedback to figure how to better make use of the feature.

Add to release note stuff like '20 threads by default'. I'd be +1 for having it on by default but 20 seems like a high number. 5 threads by default?

Reply:
> Stats on executors for general RS sounds good. Need feedback to figure how to better make use of the feature.
> 
> Add to release note stuff like '20 threads by default'. I'd be +1 for having it on by default but 20 seems like a high number. 5 threads by default?

Since ExecutorService#getAllExecutorStatuses will return all executor’s status info, I think It‘s not difficult to collect them into metrics system, will track this in HBASE-23083

Reply:
I think the feature is more suitable for low concurrent but high batch size scenarios(the number of multiget requests is small,  but the batch size of each multiget is very large).
If the number of  RS Active handlers is larger than the capacity of the parallelGet threadpool, there should be performance degradation instead, So it's better not to enable it by default? 
will explain this in releasenote.